src/main/java/org/jitsi/videobridge/octo/OctoTransportManager.java

```diff
@@ -0,0 +1,459 @@
+    private void initializeSockets()
+        throws IOException
+    {
+        synchronized (socketsSyncRoot)
+        {
+            if (rtpSocket != null)
+            {
+                //rtpSocket and rtcpSocket are always set together
+                return;
+            }
+
+            MultiplexingDatagramSocket relaySocket = octoRelay.getSocket();
+
+            rtpSocket
+                = createOctoSocket(
+                    relaySocket.getSocket(channel.getDatagramFilter(false)));
+            rtcpSocket
+                = createOctoSocket(
+                    relaySocket.getSocket(channel.getDatagramFilter(true)));
+        }
+    }
+
+    /**
+     * Creates a {@link DatagramSocket} instance which wraps a specific other
+     * {@link DatagramSocket} instance for the purposes of Octo. Specifically,
+     * it overrides {@code receive()} to strip the Octo headers, and overrides
+     * {@code send()} to send the packet to the addresses of all remote relays.
+     *
+     * @param socket the socket to be wrapped.
+     * @return the wrapped socket.
+     * @throws SocketException
+     */
+    private DatagramSocket createOctoSocket(DatagramSocket socket)
+        throws SocketException
+    {
+        return new DelegatingDatagramSocket(socket)
+        {
+            @Override
+            public void receive(DatagramPacket p)
+                throws IOException
+            {
+                super.receive(p);
+
+                try
+                {
+                    // Strip the Octo header. The code which actually handles
+                    // the values in the Octo header has already executed in
+                    // the accept() method of the DatagramPacketFilter for the
+                    // associated channel.
+                    p.setData(
+                            p.getData(),
+                            p.getOffset() + OctoPacket.OCTO_HEADER_LENGTH,
+                            p.getLength() - OctoPacket.OCTO_HEADER_LENGTH);
+                }
+                catch (Exception e)
+                {
+                    // TODO: more graceful handling
+                    logger.error(
+                        "Failed to strip Octo header while receiving a packet:"
+                            + e);
+                }
+            }
+
+            @Override
+            public void send(DatagramPacket p)
+                throws IOException
+            {
+                doSend(p);
+            }
+        };
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public MediaStreamTarget getStreamTarget(Channel channel)
+    {
+        // We'll just hack in the local address so that we have something
+        // non-null. It should never be used anyway.
+        DatagramSocket socket = octoRelay.getSocket();
+        InetAddress inetAddress = socket.getLocalAddress();
+        int port = socket.getLocalPort();
+        return new MediaStreamTarget(inetAddress, port, inetAddress, port);
```

Comment:
should it re-create this every call?

Reply:
Yeah, MediaStreamTarget is just a container for a pair of InetSocketAddress. The method is only called once when the media stream starts, so it's not worth caching